application/src/main/java/org/thingsboard/server/service/install/SqlDatabaseUpgradeService.java

```diff
@@ -222,6 +231,25 @@ public void upgradeDatabase(String fromVersion) throws Exception {
                     schemaUpdateFile = Paths.get(installScripts.getDataDir(), "upgrade", "3.1.0", SCHEMA_UPDATE_SQL);
                     loadSql(schemaUpdateFile, conn);
                     log.info("Schema updated.");
+                    try {
+                        conn.createStatement().execute(
+                                "ALTER TABLE device ADD entity_profile_id varchar(31);"); //NOSONAR, ignoring because method used to execute thingsboard database upgrade script
```

Comment:
Do we have a constraint for deletion of entity-profile if it is already used for some devices? We should add one.

Reply:
Maybe is it better to create foreign key for entity_profile_id after setting default profiles?